src/inference/src/cpp/ie_plugin.hpp

```diff
@@ -196,7 +200,11 @@ class InferencePlugin {
     }
 
     SoPtr<ie::IExecutableNetworkInternal> compile_model(const std::string& modelPath, const std::map<std::string, std::string>& config) {
-        OV_PLUGIN_CALL_STATEMENT(return {_ptr->LoadNetwork(modelPath, config), _so});
+        SoPtr<ie::IExecutableNetworkInternal> res;
+        OV_PLUGIN_CALL_STATEMENT(res = _ptr->LoadNetwork(modelPath, config));
+        if (!res._so)
+            res._so = _so;
```

Comment:
why do we override `_so` here?

Reply:
currently only auto/multi override this LoadNetwork(modelPath***) function, so I have to return SoExecutableNetworkInternal there also. So I add a checker here to check if the res._so is the placeholder so from multi plugin or the real so from core->LoadNetwork(cnnnetwork**)